blablahelp-backend/src/main/java/org/nnn4eu/hfische/blablahelp/blablahelpbackend/userdata/UserDataService.java

```diff
@@ -0,0 +1,46 @@
+package org.nnn4eu.hfische.blablahelp.blablahelpbackend.userdata;
+
+import lombok.RequiredArgsConstructor;
+import org.nnn4eu.hfische.blablahelp.blablahelpbackend.account.AccountService;
+import org.nnn4eu.hfische.blablahelp.blablahelpbackend.shared.model.Address;
+import org.nnn4eu.hfische.blablahelp.blablahelpbackend.shared.model.AddressWrap;
+import org.nnn4eu.hfische.blablahelp.blablahelpbackend.shared.model.EAddressType;
+import org.nnn4eu.hfische.blablahelp.blablahelpbackend.userdata.model.Offer;
+import org.nnn4eu.hfische.blablahelp.blablahelpbackend.userdata.model.UserData;
+import org.springframework.stereotype.Service;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class UserDataService {
+    private final UserDataRepo userDataRepo;
+    private final OfferRepo offerRepo;
+    private final AccountService accountService;
+
+    public UserData findUserDataById(@NotBlank String accountId){
+        boolean ok=accountService.accountExistsById(accountId);
+        if(!ok)throw new IllegalArgumentException("account with given id doesn't exist");
+        return userDataRepo.findById(accountId).orElseGet(()->userDataRepo.save(new UserData(accountId)));
+    }
```

Comment:
Wait - what?!
You are throwing an exception because an id wasn't found and at the same time you are creating a new account with the not found id?
In my world only one of this two actions takes place.🤔